packages/core/src/app/checkout/Checkout.test.tsx

```diff
@@ -0,0 +1,328 @@
+                }),
+            ).toBeInTheDocument();
+            expect(screen.getByText(/shipping method/i)).toBeInTheDocument();
+        });
+
+        it('logs unhandled error', async () => {
+            const error = new Error();
+
+            jest.spyOn(checkoutService, 'loadShippingAddressFields').mockImplementation(() => {
+                throw error;
+            });
+
+            checkout.use('CartWithBillingEmail');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForShippingStep();
+
+            expect(defaultProps.errorLogger.log).toHaveBeenCalledWith(error);
+        });
+    });
+
+    describe('billing step', () => {
+        it('renders billing component when billing step is active', async () => {
+            checkout.use('CartWithShippingAddress');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForBillingStep();
+
+            expect(
+                screen.getByRole('textbox', {
+                    name: /address/i,
+                }),
+            ).toBeInTheDocument();
+            expect(screen.getByText(/billing address/i)).toBeInTheDocument();
+        });
+
+        it('renders shipping component with summary data', async () => {
+            checkout.use('CartWithShippingAddress');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForBillingStep();
+
+            expect(screen.getByText(/new south wales,/i)).toBeInTheDocument();
+            expect(screen.getByText(/pickup in store/i)).toBeInTheDocument();
+        });
+
+        it('logs unhandled error', async () => {
+            const error = new Error();
+
+            jest.spyOn(checkoutService, 'loadBillingAddressFields').mockImplementation(() => {
+                throw error;
+            });
+
+            checkout.use('CartWithShippingAddress');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForBillingStep();
+
+            expect(defaultProps.errorLogger.log).toHaveBeenCalledWith(error);
+        });
+    });
+
+    describe('payment step', () => {
+        it('renders payment component when payment step is active', async () => {
+            checkout.use('CartWithShippingAndBilling');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForPaymentStep();
+
+            expect(screen.getByText(/test payment provider/i)).toBeInTheDocument();
+            expect(screen.getByText(/pay in store/i)).toBeInTheDocument();
+            expect(screen.getByText(/place order/i)).toBeInTheDocument();
+        });
+
+        it('logs unhandled error', async () => {
+            checkout.use('CartWithShippingAndBilling');
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForPaymentStep();
+
+            expect(defaultProps.errorLogger.log).toHaveBeenCalled();
+        });
```

Comment:
Payment step initialization in local Node environment will result in the error shown below, thus no error mockup is required.
```
[InvalidArgumentError]: The provided URL must be absolute.
```